### ISSUE TITLE
Confusing deprecation warnings

### ISSUE THREAD

**Issue by Akaricchi** [mesonbuild/meson#4716]
```
DEPRECATION: Project targetting '>=0.45.0' but tried to use feature deprecated since '0.48.0': python3 module
```

Since I'm targeting an old version where the python3 module is not yet deprecated, and Meson is aware of that, should I really get a warning like this? The wording makes it sound like I'm doing something wrong, yet the opposite is true: *not* using the deprecated feature would've been the wrong thing to do, as it would break compatibility with my target version.

Furthermore, it seems like this system is not universally applied. The following warning has not changed since its introduction:

```
DEPRECATION: build_always is deprecated. Combine build_by_default and build_always_stale instead.
```

i.e. it doesn't know which versions it applies to.

**Comment by jeremiah** [comment]
The warning itself is misspelled: targeting should be spelled targeting, not targetting.

**Comment by eli-schwartz** [comment]
A duplicate issue was opened in #7080 and this then got fixed by #7083
